sdks/python/apache_beam/typehints/opcodes.py

```diff
@@ -188,6 +204,31 @@ def store_subscr(unused_state, unused_args):
   pass
 
 
+def binary_slice(state, args):
+  _ = state.stack.pop()
+  _ = state.stack.pop()
+  base = Const.unwrap(state.stack.pop())
+  if base is str:
+    out = base
+  elif isinstance(base, typehints.IndexableTypeConstraint):
+    out = base
+  else:
+    out = element_type(base)
```

Comment:
For my understanding, what is the interpretation of
`out = element_type(base)`?

does this say: take  a container, and return the type of individual element that lives in the container? Wouldn't a slice always return a container as opposed to individual elements?

Reply:
Hmm that's actually an implementation interpretation I hadn't thought of. The documentation for the op is here - https://docs.python.org/3/library/dis.html#opcode-BINARY_SLICE and I was adapting the BINARY_SUBSCR code (which is doing similar work with a single index arg instead of a slice.) I think most cases will be caught by the indexable type constraint (so the type of the container is what we return) but that else case might not be handling things correctly. I'll see if I can get a bit more of an understanding on this op code specifically.